examples/siamese_network/siamese_network.py

```diff
@@ -0,0 +1,284 @@
+        than that of the anchor image (named as negative_image). We return the anchor image,
+        positive image, negative image and anchor label.
+        """
+
+        # obtain the anchor image
+        anchor_image = self.data[index].clone().float()
+
+        # obtain the class label of the anchor image
+        anchor_label = self.dataset.targets[index].clone().float()
+
+        # pick some random class for the first image
+        selected_class = random.randint(0, 9)
+
+        # pick a random index for the first image in the grouped indices based of the label
+        # of the class
+        random_index_1 = random.randint(0, self.grouped_examples[selected_class].shape[0] - 1)
+
+        while random_index_1 != anchor_label:
+            random_index_1 = random.randint(0, self.grouped_examples[selected_class].shape[0] - 1)
+
+        # pick the index to get the first image
+        index_1 = self.grouped_examples[selected_class][random_index_1]
+
+        # get the first image
+        positive_image = self.data[index_1].clone().float()
+
+        random_index_2 = random.randint(0, self.grouped_examples[selected_class].shape[0] - 1)
+
+        # ensure that the index of the second image isn't the same as the first image
+        while random_index_2 == anchor_label:
+            random_index_2 = random.randint(0, self.grouped_examples[selected_class].shape[0] - 1)
+
+        # pick the index to get the second image
+        index_2 = self.grouped_examples[selected_class][random_index_2]
+
+        # get the second image
+        negative_image = self.data[index_2].clone().float()
+
+        return anchor_image, positive_image, negative_image, anchor_label
+
+
+def train(model, device, optimizer, train_loader, lr_scheduler, log_interval, max_epochs):
+
+    criterion = nn.TripletMarginLoss()
+
+    # define model training step
+    def train_step(engine, batch):
+        model.train()
+        anchor_image, positive_image, negative_image, anchor_label = batch
+        anchor_image = anchor_image.to(device)
+        positive_image, negative_image = positive_image.to(device), negative_image.to(device)
+        anchor_label = anchor_label.to(device)
+        optimizer.zero_grad()
+        anchor_out = model(anchor_image)
+        positive_out = model(positive_image)
+        negative_out = model(negative_image)
+        loss = criterion(anchor_out, positive_out, negative_out)
+        loss.backward()
+        optimizer.step()
+        return loss
+
+    # create a trainer engine and attach train_step
+    trainer = Engine(train_step)
+
+    # attach progress bar to trainer
+    pbar = ProgressBar()
+    pbar.attach(trainer)
+
+    # attach various handlers to trainer engine
+    @trainer.on(Events.ITERATION_COMPLETED(every=log_interval))
+    def log_training_results(engine):
+        print(f"Train Epoch: {engine.state.epoch}, Train Loss: {engine.state.output: .5f}")
```

Comment:
If you are using progress bar, I think there is no need to print loss. You can do the following:
```python
    pbar = ProgressBar()
    pbar.attach(trainer, metric_names=['loss'])
```

Reply:
Oh alright, I will add it